storage/filesystem/object.go

```diff
@@ -545,6 +545,9 @@ func (s *ObjectStorage) HashesWithPrefix(prefix []byte) ([]plumbing.Hash, error)
 
 	// TODO: This could be faster with some idxfile changes,
 	// or diving into the packfile.
```

Comment:
This comment is related to the `for` loop, therefore the changes must be placed before it.

Reply:
I think the comment applies to both sections together, if the code was switched to dive into packfile then it wouldn't require populating the index?
(I'm still happy to move it if you still think its better between the population and usage)